dvc/remote/base.py

```diff
@@ -33,14 +33,16 @@ def __init__(self, msg):
 
 class RemoteBase(object):
     REGEX = None
+    REQUIRES = []
 
     def __init__(self, project, config):
         pass
 
     @classmethod
-    def supported(cls, config):
+    def supported(cls, config, check_dependencies=False):
         url = config[Config.SECTION_REMOTE_URL]
-        return cls.match(url) != None
+        deps_ok = not check_dependencies or all(cls.REQUIRES)
```

Comment:
It would be great if we could print a warning here that the url is supported but the optional dependency is missing.

Reply:
Makes sense. It will raise an assertion error during schema validation that ultimately says that `supported_url(...) should return True`, but if a logging message right before says "URL is valid but required dependencies to use it are missing".